packages/assets-controllers/src/AssetsContractController.ts

```diff
@@ -688,6 +702,78 @@ export class AssetsContractController {
     }
     return nonZeroBalances;
   }
+
+  async getStakedBalanceForChain(
+    address: string,
+    networkClientId?: NetworkClientId,
+  ): Promise<StakedBalance> {
+    const chainId = this.#getCorrectChainId(networkClientId);
+    const provider = this.#getCorrectProvider(networkClientId);
+
+    // balance defaults to zero
+    let balance: BigNumber = BigNumber.from(0);
+
+    // Only fetch staked balance on supported networks
+    if (
+      ![
+        SupportedStakedBalanceNetworks.mainnet,
+        SupportedStakedBalanceNetworks.holesky,
+      ].includes(chainId as SupportedStakedBalanceNetworks)
+    ) {
+      return undefined as StakedBalance;
+    }
+    // Only fetch staked balance if contract address exists
+    if (
+      !((id): id is keyof typeof STAKING_CONTRACT_ADDRESS_BY_CHAINID =>
+        id in STAKING_CONTRACT_ADDRESS_BY_CHAINID)(chainId)
+    ) {
+      return undefined as StakedBalance;
+    }
+
+    const contractAddress = STAKING_CONTRACT_ADDRESS_BY_CHAINID[chainId];
+    const abi = [
+      {
+        inputs: [{ internalType: 'address', name: 'account', type: 'address' }],
+        name: 'getShares',
+        outputs: [{ internalType: 'uint256', name: '', type: 'uint256' }],
+        stateMutability: 'view',
+        type: 'function',
+      },
+      {
+        inputs: [{ internalType: 'uint256', name: 'shares', type: 'uint256' }],
+        name: 'convertToAssets',
+        outputs: [{ internalType: 'uint256', name: 'assets', type: 'uint256' }],
+        stateMutability: 'view',
+        type: 'function',
+      },
+    ];
+
+    try {
+      const contract = new Contract(contractAddress, abi, provider);
+      const exchangeRateDenominator = BigNumber.from((1e18).toString());
+      const multiplier = exchangeRateDenominator;
+      const userShares = await contract.getShares(address);
+
+      // convert shares to assets only if address shares > 0 else return default balance
```

Comment:
Do we need to calculate exchange rate here or can we directly get assets from convertToAssets function ?

Reply:
Hm, you're right we don't really need to get the rate here. Thx will make that change!